src/components/Modal.js

```diff
@@ -0,0 +1,92 @@
+import "../polyfills/closest.polyfill";
+import "./w3-dialog";
+import { GetFirstElementOrDefault } from "../utilities/dom.utils";
+
+const cssClasses = {
+  active: "active",
+  modalCloseButton: "dg_modal__close-button",
+  modalOpenButton: "dg_modal__open-button",
+  modalOverlay: "dialog-backdrop",
+  /** overlay is transparent gray section around the modal */
+  canCloseOnOverlayClick: "can-dismiss"
+};
+
+const selectors = {
+  activeDismissibleModal: ".dg_modal[data-dismissible]"
+};
+
+/**
+ * Handle the modal overlay click.
+ * The modal markup will determine whether or not the modal can close on overlay click
+ * based on whether it has the data-dismissible data attribute.
+ * @param {element:click} clickEvent - the observable overlay click event
+ * @listens element:click
+ */
+const handleActiveOverlayClick = clickEvent => {
+  const activeDismissibleModalElm = GetFirstElementOrDefault(
+    document,
+    selectors.activeDismissibleModal
+  );
+  const isDismissible =
+    activeDismissibleModalElm &&
+    activeDismissibleModalElm.getAttribute("data-dismissible") !== "false";
+
+  if (isDismissible) {
+    /** The  w3 logic we used requires the close button to be passed into the closeDialog function */
+    const closeButton = GetFirstElementOrDefault(
+      activeDismissibleModalElm,
+      `.${cssClasses.modalCloseButton}`
+    );
+
+    window.closeDialog(closeButton);
+  }
+};
+
+/**
+ * Captures all dom click events in order to bind all event with a particular class
+ * @param {document:click} clickEvent - the observable click event
+ * @listens document:click
+ */
+const handleDocumentClick = clickEvent => {
+  const { classList: targetClassList } = clickEvent.target;
+  // If the clicked element doesn't have the right selector, bail
+  const isModalOpenButtonClick = targetClassList.contains(
+    cssClasses.modalOpenButton
+  );
+  const isModalCloseButtonClick =
+    targetClassList.contains(cssClasses.modalCloseButton) ||
+    !!clickEvent.target.closest(`.${cssClasses.modalCloseButton}`);
+  const isOverlayClick =
+    targetClassList.contains(`${cssClasses.modalOverlay}`) &&
+    targetClassList.contains(`${cssClasses.active}`);
+
```

Comment:
Cant 60 and 61 just be targetClassList.contains(cssClasses.modalOverlay) like you have on line 53?

Reply:
No I don't think this line is saying, the target must have the modal overlay class, and the overlay must be active or visible.

Does that make sense?

Reply:
Right but what Im saying is why add the template literal to that? Isnt the value for cssClass.active or cssClass.modalOverlay already a string? Like on line 53 you have targetClassList.contains( cssClasses.modalOpenButton) without the literal and its going to check for the string value without it so is there a reason why you added that for 60 and 61?

Reply:
@tmccoy529 i see what you are saying now. Will update.

Reply:
removed, really good catch.